### PR TITLE
fix(dashboard): remove horizontal scrolling for long-titled dashboards & allow showing dashboard title in 2 lines

### PR DESCRIPTION
**Pull request by artemonsh**
### SUMMARY

1. For dashboards with long title all elements don't fit to screen anymore and horizontal scrolling appears. This PR is an attempt to solve this issue.
2. I suggest to show not 1 but 2 lines of the title if it is long enough (which is a common practice in my company). The overflow will be hidden as it was before. It is done with the help of CSS properties webkit-line-clamp and webkit-box-orient (example [here](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp)). Moreover, it is looks so much better on smartphone.

It works as expected in Chrome and Firefox, I did not check the behavior in other browsers.

### BEFORE
image

### AFTER
image

### BEFORE vs AFTER video (with sound)
https://user-images.githubusercontent.com/53895552/211523820-fe913c9f-d387-450c-ada0-72a1ba7572ee.mp4

### TESTING INSTRUCTIONS

Create a dashboard with a very long title. You can take it from this website: https://www.lipsum.com

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/22520
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ type DashboardBuilderProps = {};
 
 const StyledDiv = styled.div`
   display: grid;
-  grid-template-columns: auto 1fr;
+  grid-template-columns: auto minmax(0, 1fr);
```
Comment (artemonsh) [inline superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:86]: This is the only change to fix horizontal scrolling

**Comment by artemonsh** [comment]
@villebro could you please take a look?

**Comment by rusackas** [comment]
Thanks for the fantastic PR description/videos (narration is such an amazing and unusual thing around here!). Also, sorry this slid under our collective radar for so long. Are you able to give this a quick rebase to bring it up to a mergeable state?

Also, I haven't checked the caniuse.com entries for these css properties yet, but I'm assuming these are pretty friendly for most browsers by now and might not need the webkit vendor prefix anymore if we're lucky :D

**Comment by rusackas** [comment]
@artemonsh this could use a rebase now. Let me know if you intend to revisit this, otherwise, we can convert it to a draft if you need a while, or close it if it's a lost cause. Thanks in any case :)

**Comment by artemonsh** [comment]
@rusackas  I don't have time for this, let's close it and hope someone else will finish it

**Pull request closed without merging**